File: arrangeOddEvenNums.py
class Solution:
    def reOrderArray(self, array):
        odd = []
        even = []

        for _ in array:
            if _%2 == 0:
                even.append(_)
            else:
                odd.append(_)
        return odd+even
            
if __name__ == "__main__":
    array = [1,2,3,4,5,6,7]
    s = Solution()
    print s.reOrderArray(array)

File: arrangeOddEvenNums.cpp
#include <iostream>
#include <vector>
using namespace std;

class Solution{
public:
    void reOrderArray(vector<int>& array){
        if (array.size() <= 1)
            return ;
        int i = -1;
        int j = i+1;
        while (true){
            while (array[++i] % 2 != 0){
                if (i == array.size()-1)
                    break;
            }

            while (array[++j] % 2 == 0){
                if (j == array.size()-1)
                    break;
            }

            if (i >= j)
                break;
            swap(array[i], array[j]);

            if (i == array.size()-1 || j == array.size()-1)
                break;
        }
    }

};

int main(){

    int A[] = {2,4,6,1,3,5};
    vector<int> nums(&A[0], &A[6]);
    Solution s;
    s.reOrderArray(nums);
    for(int i = 0; i < nums.size(); i++){
        cout << nums[i] << " ";
    }
}
